### PR TITLE
chore: wire up profiling configurations [DET-5064]

### PR DESCRIPTION
**Pull request by stoksc**
## Description
This change wires up the profiling configurations in the master. The exact configurations added are described by the observability ERD.

## Test Plan
- [x] add a test
- [x] make sure `print(experiment_config)` in harness.py show the configs.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: docs/reference/experiment-config.txt [hunk 1]
```diff
@@ -1223,6 +1223,27 @@ for more details on how to set environment variables in containers.
    The file system path to use as the mount point in the trial runner
    container for storing the local cache.
 
+***********
```
Comment (stoksc) [inline docs/reference/experiment-config.txt:1226]: This is WIP, I won't land without cleaning it up/clarifying based on what we actually land in the other PRs.

**Review by armandmcqueen — COMMENTED** [review]
I think you need to integrate this into the while experiment config schema thing

File: docs/reference/experiment-config.txt [hunk 2]
```diff
@@ -1223,6 +1223,27 @@ for more details on how to set environment variables in containers.
...
+experiment profiling. Determined currently supports the collection of
+system metrics for all frameworks (up to 5 minutes of collection) and
+the collection of timing metrics for PyTorchTrial.
+
+**Optional Fields**
+
+``enabled``
+   Specifies if profiling is enabled for the given experiment.
+
+``window``
```
[the remark this answers is not in the document]
Comment (stoksc) [reply docs/reference/experiment-config.txt:1240]: @katport you have thoughts? I know this is what we agreed on in the ERD review, I think with your input factored in.

**Comment by stoksc** [comment]
> To me this is a sort of complicated way to represent this information. I would rather we don't have a custom string format for "window" that we need to parse.

We're good to land like this, @rb-determined-ai is aware.

File: docs/reference/experiment-config.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (stoksc) [reply docs/reference/experiment-config.txt:1240]: @armandmcqueen i think i explicitly specified something like this in the original erd review and it got shot down. i don't think the custom format is hard on the backend, do you think it's tricky for the user?

**Comment by rb-determined-ai** [comment]
> We're good to land like this, @rb-determined-ai is aware

Lol leave me out of this.  I didn't look that closely.

The thing that gives me pause is that I don't think we string-parse any entries of the experiment config anywhere else.  Not sure if that's the right pattern, but it's definitely the pattern we have.

Also, FWIW this is from the Profiling section of the docstring from Keras' TensorBoard callback:

  ```python
  # Profile a single batch, e.g. the 5th batch.
  tensorboard_callback = tf.keras.callbacks.TensorBoard(
      log_dir='./logs', profile_batch=5)
  model.fit(x_train, y_train, epochs=2, callbacks=[tensorboard_callback])

  # Profile a range of batches, e.g. from 10 to 20.
  tensorboard_callback = tf.keras.callbacks.TensorBoard(
      log_dir='./logs', profile_batch=(10,20))
  model.fit(x_train, y_train, epochs=2, callbacks=[tensorboard_callback])
  ```

**Comment by stoksc** [comment]
@armandmcqueen @rb-determined-ai oops lol, i quoted the wrong thing, I mean

> I think you need to integrate this into the while experiment config schema thing

We're good to land like this, @rb-determined-ai is aware.

lol

**Review by armandmcqueen — APPROVED** [review]
🚀
